File: python/dp-e08.py

count_bottles = "{} bottles of beer on the wall, {} bottles of beer."
remove_bottle = "Take one down and pass it around, {} bottles of beer on the wall."
no_more = "No more bottles of beer on the wall, no more bottles of beer."
finisher = "Go to the store and buy some more, 99 bottles of beer on the wall."

if __name__ == '__main__':
    for b in range(1, 100)[::-1]:
        print(count_bottles.format(b, b))
        print(remove_bottle.format(b-1 if b-1>0 else 'no more'))

    print(no_more)
    print(finisher)

File: javascript/dp-e08.js

function bottlesSong () {
  for (let i=100; i >= 1; i--) {
    console.log(`${i} bottles of beer on the wall, ${i} bottles of beer.`);
    console.log(`Take one down and pass it around, ${i-1>0 ? i-1 : 'no more'} bottles of beer on the wall.`);
  }
  console.log("No more bottles of beer on the wall, no more bottles of beer.");
  console.log("Go to the store and buy some more, 99 bottles of beer on the wall.");
}

bottlesSong();
